### PR TITLE
clean channel for charmcraft publish from disallowed signs

### PR DESCRIPTION
**Pull request by beliaev-maksim**

### CODE REVIEW

**Comment by simskij** [comment]
This does not look to me like something the action should do, to be honest. Can you please explain why this is needed? My view is that the user needs to make sure they pass valid channel names to the action, rather than the action trying to rectify it for them.

**Comment by beliaev-maksim** [comment]
overall, charmhub has its own set of restrictions on the name. One of the use cases is that you publish charm to the branch on charmhub with the same branch name as your git branch. However, git branches have more flexibility on names, for example `_` which is not supported on charmhub. So, we can just filter down such things in action. Otherwise we need to do it in all the repos

File: src/services/charmcraft/charmcraft.ts [hunk 1]
```diff
@@ -193,12 +193,14 @@ class Charmcraft {
     const globber = await glob.create('./*.charm');
     const paths = await globber.glob();
 
+    // filter all characters which are not letters, numbers or hyphens
+    const allowedChannel = channel.replace(/_|[^\w\-/]/gi, '');
```
Comment (simskij) [inline src/services/charmcraft/charmcraft.ts:197]:
```suggestion
    const allowedChannel = channel.replace(/[^a-zA-Z0-9\-\/]/gi, '');
```

**Review by simskij — CHANGES_REQUESTED** [review]
Suggest switching to an explicit regex construct to make it easier for others to parse.

File: src/services/charmcraft/charmcraft.ts [hunk 2]
```diff
@@ -194,7 +194,7 @@ class Charmcraft {
     const paths = await globber.glob();
 
     // filter all characters which are not letters, numbers or hyphens
-    const allowedChannel = channel.replace(/_|[^\w\-/]/gi, '');
+    const allowedChannel = channel.replace(/[^a-zA-Z0-9\-\/]/gi, '');
```
Comment (simskij) [inline src/services/charmcraft/charmcraft.ts:197]:
```suggestion
    const allowedChannel = channel.replace(/[^a-zA-Z0-9\-/]/gi, '');
```

**Review by simskij — APPROVED** [review]

**Comment by ca-scribner** [comment]
This was reverted by #93

**Pull request merged**